### PR TITLE
Implement Container Stats Streaming

### PR DESCRIPTION
**Pull request by casperisfine**
I needed it, and I saw the TODO.

@tlunter thoughts?

### CODE REVIEW

File: spec/docker/container_spec.rb [hunk 1]
```diff
@@ -84,22 +84,40 @@
...
+
+    context "when streaming container stats" do
+      subject {
+        described_class.create('Cmd' => ['sleep', '1'], 'Image' => 'debian:wheezy')
+      }
+
+      it "yields a Hash" do
+        subject.start! # If the container isn't started, no stats will be streamed
+        subject.stats do |output|
+          expect(output).to be_a Hash
```
Comment (tlunter) [inline spec/docker/container_spec.rb]: We should set a variable in here and make an expectation that it got here outside the block. Also should we validate it got hit some number of times?

Comment (casperisfine) [inline spec/docker/container_spec.rb]: Makes sense.

Comment (casperisfine) [inline spec/docker/container_spec.rb]: Done. 

NB: The API is kinda funky as if the container stop, the connection isn't closed, and we keep trying to read it forever. Not sure what we could do about it from the client side.

**Comment by casperisfine** [comment]
I've updated the PR again based on my experimentations with this branch.

So basically the Docker daemon won't ever close a `stats` stream, so we have to rely on Excon read timeout to know when there won't be any more stats for a container.

File: lib/docker/container.rb [hunk 2]
```diff
@@ -158,9 +158,22 @@ def logs(opts = {})
...
+    if block_given?
+      options[:read_timeout] ||= 10
+      options[:idempotent] ||= false
+      parser = lambda do |chunk, remaining_bytes, total_bytes|
+        yield Docker::Util.parse_json(chunk)
+      end
+      begin
+        connection.get(path_for(:stats), nil, {response_block: parser}.merge(options))
+      rescue Docker::Error::TimeoutError
+        # If the container stops, the docker daemon will hold the connection open forever, but stop sending events.
```
Comment (tlunter) [inline lib/docker/container.rb]: Can you just break up this comment? Otherwise looks good.

Comment (casperisfine) [inline lib/docker/container.rb]: Done.

**Pull request merged**